### PR TITLE
21 profile page update email addres

### PR DESCRIPTION
**Pull request by halimyu**
### Navigation and Layout Enhancements:
- Introduced a fixed navigation bar at the left side of settings page.
- Adjusted overall styling for improved alignment and spacing consistency.

### Settings Page Enhancements:
- Created a **Settings page** under the `SettingsController`.
- Added the ability to update the user's **name** and display the current email.
- Implemented a modal popup for changing the user's email:
  - Validates current password.
  - Verifies new email and confirmation match before saving changes.

---

## Testing
- Tested the modal-based email update process:
  - Validates current password, email confirmation, and error handling.

---

## Notes

- These changes significantly enhance the user interface and navigation experience.

---

image
image
image

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -121,6 +121,43 @@ $body-color: $text-default;
   vertical-align: middle;
 }
 
+.nav-pills {
```
Comment (bjohnson66) [inline app/assets/stylesheets/application.scss:124]: conflicts averted

File: app/controllers/settings_controller.rb [hunk 2]
```diff
@@ -1,5 +1,38 @@
 class SettingsController < ApplicationController
+  before_action :set_current_user
```
Comment (bjohnson66) [inline app/controllers/settings_controller.rb:2]: yes, make sure user is logged in

File: app/controllers/settings_controller.rb [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
+    # Update the email without triggering validations
+    if @current_user.update_columns(email: params[:new_email]) # Skips validations
+      flash[:success] = "Your email has been updated successfully."
+      redirect_to settings_path
+    else
+      flash[:danger] = "Failed to update email. Please try again."
+      redirect_to settings_path
+    end
+  end
+
```
Comment (bjohnson66) [inline app/controllers/settings_controller.rb:37]: flash messages all look good

File: config/routes.rb [hunk 4]
```diff
@@ -26,6 +26,7 @@
 
   get '/landing', to: 'landing#index', as: 'landing'
   get 'settings', to: 'settings#settings', as: 'settings'
+  patch 'change_email', to: 'settings#change_email'
```
Comment (bjohnson66) [inline config/routes.rb:29]: lgtm

**Review by bjohnson66 — APPROVED** [review]
Ran on my machine, all tests pass

**Review by zhasa18 — APPROVED** [review]
Good job!

**Pull request merged**